Patch merged in chromium already.
Reviewed at: https://codereview.chromium.org/295333002/ (bug: crbug.com/377085)

diff --git a/base/md5.cc b/base/md5.cc
index e77885e..6227ee6 100644
--- a/base/md5.cc
+++ b/base/md5.cc
@@ -251,8 +251,12 @@ void MD5Final(MD5Digest* digest, MD5Context* context) {
         byteReverse(ctx->in, 14);
 
         /* Append length in bits and transform */
-        ((uint32 *)ctx->in)[ 14 ] = ctx->bits[0];
-        ((uint32 *)ctx->in)[ 15 ] = ctx->bits[1];
+        memcpy(&ctx->in[14 * sizeof(ctx->bits[0])],
+               &ctx->bits[0],
+               sizeof(ctx->bits[0]));
+        memcpy(&ctx->in[15 * sizeof(ctx->bits[1])],
+               &ctx->bits[1],
+               sizeof(ctx->bits[1]));
 
         MD5Transform(ctx->buf, (uint32 *)ctx->in);
         byteReverse((unsigned char *)ctx->buf, 4);
